Robot2019/src/main/java/frc/robot/commands/Climb.java

```diff
@@ -63,4 +70,27 @@ protected void end() {
   protected void interrupted() {
     end();
   }
+
+  private void climb() {
+    dt.drive(-0.5, -0.5);
+    if (up) {
+      climber.runClimber(0.5);
+      if (!climber.needToClimb())
+        up = false;
+    } else {
+      climber.runClimber(-0.5);
+      if (!climber.canDrop())
+        up = true;
+    }
+  }
+
+  private void retract() {
+    climber.retractClimber();
+  }
```

Comment:
either keep both the climb() and retract() logic all in the Climber subsystem class or all in here, preferably in the subsystem